services/scrape-test/test_samples.py

```diff
@@ -0,0 +1,221 @@
+                for iframe in iframe_elements:
+                    child_frame = iframe.content_frame()
+                    if child_frame:
+                        result = find_youtube_in_frames([child_frame])
+                        if result:
+                            return result
+            return None
+
+        # Start the search from the main page frames
+        youtube_url = find_youtube_in_frames(page.frames)
+
+        if youtube_url:
+            return youtube_url
+        else:
+            print("No YouTube iframe found.")
+            return None
+
+    except Exception as e:
+        print(f"An error occurred while scraping {url}: {e}")
+        return "timeout_error"
+    finally:
+        page.close()
+
+
+def convert_embed_to_watch_url(embed_url: str) -> str:
+    """
+    Convert a YouTube embed URL to a standard watch URL.
+
+    Args:
+        embed_url (str): The YouTube embed URL.
+
+    Returns:
+        str: The standard YouTube watch URL.
+    """
+    # Regular expression to extract the video ID from the embed URL
+    match = re.search(r"/embed/([^?&]+)", embed_url)
+    if match:
+        video_id = match.group(1)
+        watch_url = f"https://www.youtube.com/watch?v={video_id}"
+        return watch_url
+    else:
+        # If the embed URL doesn't match the expected pattern, return None or handle accordingly
+        print(f"Invalid embed URL format: {embed_url}")
+        return None
+
+
+def main():
+    # Step 1: Fetch rows from BigQuery
+    rows = get_rows_from_bq()
+
+    # Counters
+    total_urls_processed = 0
+    successful_scrapes = (
+        0  # Number of YouTube links found and inserted into original table
+    )
+    timeouts_inserted = 0  # Number of timeouts inserted into new table
+
+    # Initialize Playwright and the browser
+    with sync_playwright() as pw:
+        browser = pw.chromium.launch(headless=True)  # Set to False for debugging
+        context = browser.new_context()
+
+        # Step 3: Scrape each URL for YouTube links
+        for row in rows:
+            # Stop the script when 50 rows have been inserted into each table
+            if successful_scrapes >= 50 and timeouts_inserted >= 50:
+                print(
+                    "Reached 50 successful scrapes and 50 timeouts inserted. Exiting."
+                )
+                break
+
+            url = row.get("creative_page_url")
+            if url:
+                total_urls_processed += 1
+                print(f"\nProcessing URL {total_urls_processed}: {url}")
+                youtube_link = scrape_youtube_link(url, context)
+
+                if (
+                    youtube_link
+                    and youtube_link != "timeout_error"
+                    and "youtube.com" in youtube_link
+                ):
+                    if successful_scrapes >= 50:
+                        print(
+                            "Reached 50 successful scrapes. Skipping insertion into original table."
```

Comment:
as before move this whole function somewhere else. Also this scraping need some major refactor

Reply:
This file has been deleted now.